### DISCUSSION TITLE
Hot reloading on MDX is "slow" / doesn't exist, improvement possible?

### DISCUSSION THREAD

**Discussion by kud** [vercel/next.js#12833] [Help]
Hello,

As it must compile the whole page before refreshing it, you've got a certain lag (like 4-5s) between your editor in a .mdx file and the display on your browser.

Is there a way to accelerate this?

For instance I use vs code in .mdx + mdx preview, and it's really fast.

I know it's not the same thing as it only renders the mdx itself and not the layout in React you've got, but I'd love to have a real hot reload in mdx instead of refreshing the whole page.

Possible you think?

Thanks.

**Comment by Timer** [comment]
Could you please share a project that reproduces the 4-5s delay?

**Reply by kud** [reply]
Yes! https://github.com/kud/diary/

**Comment by kud** [comment]
Well, in fact, it's more than fine now! The beginning is a bit slow but after, the caches certainly do the thing!

Thank you and thanks for the improvements in 9.4!
